general-concepts/virtuals/text.xml

```diff
@@ -42,5 +42,113 @@ from the Gentoo repository.
 </note>
 </body>
 
+<section>
+<title>KEYWORDS in virtual packages</title>
+
+<body>
+<p>
+Since virtual packages do not install any files, they do not follow the regular
+arch testing procedure. Instead, the developer can immediately set
+their <c>KEYWORDS</c> to the union of <c>KEYWORDS</c> of their providers.
+In particular, if a new virtual is created for a stable package, the virtual is
+committed straight to stable.
+</p>
+
+<p>
+For example, if you have two packages: <c>dev-libs/liblinux</c> with
+<c>KEYWORDS="amd64 ~x86"</c> and <c>dev-libs/libbsd</c> with
+<c>KEYWORDS="~amd64-fbsd ~x86-fbsd"</c>, the resulting virtual will
+have:
+</p>
+
+<codesample lang="ebuild">
+KEYWORDS="amd64 ~x86 ~amd64-fbsd ~x86-fbsd"
+
+RDEPEND="|| ( dev-libs/liblinux dev-libs/libbsd )"
+</codesample>
+</body>
+
+</section>
+
+<section>
+<title>Virtuals and subslots</title>
+
+<body>
+<warning>
+This section is only applicable if virtual providers include versions that
+are ABI-compatible with one another (and use matching SONAMEs) and/or
+the incompatible providers are being obsoleted.
+</warning>
+
+<p>
+Alike regular packages, virtuals can define subslots that can be used
+to trigger rebuilds of their reverse dependencies. For this to work, a new
+version of the virtual is created for each subslot of the providers, and it
+depends on their specific subslots.
+</p>
+
+<p>
+For example, a virtual for different packages providing ABI-compatible
+<c>libfoo.so.1</c> library could look like the following:
+</p>
+
+<codesample lang="ebuild">
+EAPI=6
+
+DESCRIPTION="Virtual for libfoo.so.1"
+SLOT="0/1"
+
+RDEPEND="
+    || (
+        dev-libs/libfoo:0/1
+        dev-libs/libfoo-alternate:0/1
+        dev-libs/bigpack:0/libfoo-1+libbar-0
+        dev-libs/bigpack:0/libfoo-1+libbar-1
+    )
+"
+</codesample>
+
+<p>
+Virtuals can also be used when one of the providers is being obsoleted in favor
+of another but do not provide ABI-compatible versions. In this case, multiple
+versions of the virtual are created, each specifying a single provider
+and a unique subslot.
+</p>
+
+<p>
+For example, if <c>dev-libs/libfoo</c> (<c>libfoo.so.0</c>) is being replaced
+by <c>dev-libs/newfoo</c> (<c>libfoo.so.1</c>), <c>virtual/libfoo-0.ebuild</c>
+would contain:
+</p>
+
+<codesample lang="ebuild">
+EAPI=6
+
+DESCRIPTION="Virtual for libfoo.so.0"
+SLOT="0/0"
+RDEPEND="dev-libs/libfoo:0/0"
+</codesample>
+
+<p>
+while <c>virtual/libfoo-1.ebuild</c> would contain:
+</p>
+
+<codesample lang="ebuild">
+EAPI=6
+
+DESCRIPTION="Virtual for libfoo.so.1"
+SLOT="0/1"
+RDEPEND="dev-libs/newfoo:0/1"
+</codesample>
+
+<note>
+In this case, the package manager will naturally want to upgrade
+to <c>dev-libs/newfoo</c> whenever possible. Therefore, this solution
+is not viable if clean choice between the two providers is desired.
```

Comment:
Yes, you understand it correctly. I think separate paragraphs are better since the previous one describes the generic problem and solution, while this one serves as a specific introduction to the example following it.